server/auth/routes.js

```diff
@@ -7,48 +7,136 @@ const db = require("../services/database/users");
-				success: true,
-				message: "Account created",
-			});
-		})
-		.catch((err) => {
-			console.log(err);
-			next(err);
-		});
+  const { email, password, name, role } = req.body;
+
+  const user = {
+    email,
+    password,
+    name,
+    role
+  };
+
+  //   if (
+  //     user.email != null &&
+  //     user.email.length > 1 &&
+  //     user.password != null &&
+  //     user.password.length > 1 &&
+  //     user.name != null &&
+  //     user.name.length > 1 &&
+  //     user.role != null &&
+  //     user.role.length > 1 &&
+  //     (user.role == "Interpreter" || user.role == "User")
+  //   ) {
+
+  //   } else {
+  //     res.status(400).send({
+  //       success: false,
+  //       message: "Empty Validation"
+  //     });
+  //   }
+
+  if (
+    validateEmailAddress(email) &&
+    validateName(name) &&
+    validatePassword(password) &&
+    validateRole(role)
+  ) {
+    db.createUser(user)
+      .then(() => {
+        res.send({
+          success: true,
+          message: "Account created"
+        });
+      })
+      .catch(err => {
+        res.status(400).send({
+          success: false,
+          message: err
+        });
+        console.log(err);
+        next(err);
+      });
+  } else {
+    res.send(err);
+  }
+  function validateEmailAddress(email) {
+    var expression = /(?!.*\.{2})^([a-z\d!#$%&'*+\-\/=?^_`{|}~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+(\.[a-z\d!#$%&'*+\-\/=?^_`{|}~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+)*|"((([ \t]*\r\n)?[ \t]+)?([\x01-\x08\x0b\x0c\x0e-\x1f\x7f\x21\x23-\x5b\x5d-\x7e\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|\\[\x01-\x09\x0b\x0c\x0d-\x7f\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]))*(([ \t]*\r\n)?[ \t]+)?")@(([a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|[a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF][a-z\d\-._~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]*[a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])\.)+([a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|[a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF][a-z\d\-._~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]*[a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])\.?$/i;
+    if (expression.test(String(email).toLowerCase())) {
+      return true;
+    } else {
+      res.status(400).send({
+        success: false,
+        message: "The email format is incorrect."
+      });
+      return false;
+    }
+  }
+
+  //   function validateEmail(email) {
+  //     if (email == null || email.length == 0 && email.includes("@")) {
+  //       res.status(400).send({
+  //         success: false,
+  //         message: "Email required"
+  //       });
+  //       return false;
+  //     } else return true;
+  //   }
+
+  function validateName(name) {
+    if (name == null || name.length == 0) {
+      res.status(400).send({
+        success: false,
+        message: "name is required"
+      });
+      return false;
+    } else return true;
+  }
+
+  // password is invalid...
+
+  function validatePassword(password) {
+    var password_pattern = /[a-z0-9_]{8,20}/i;
```

Comment:
The validation for the password is that the length should be 8 characters minimum. You are restricting the valid characters to only letters and numbers here. Special characters can also be part of a password :) I would suggest to only check the length of the password to validate.

Reply:
Changed to check min 8 characters